packages/nimble-components/src/wafer-map/modules/experimental/computations.ts

```diff
@@ -0,0 +1,231 @@
+        );
+        this.updateScales();
+    }
+
+    public updateScales(): void {
+        const containerDiameter = Math.min(
+            this._containerDimensions.width,
+            this._containerDimensions.height
+        );
+        const gridDimensions = this.gridDimensionsValidAndDefined()
+            ? this.calculateGridDimensionsFromBoundingBox()
+            : this.calculateGridDimensionsFromDies();
+        // this scale is used for positioning the dies on the canvas
+        const originLocation = this.wafermap.originLocation;
+        this._horizontalScale = this.createHorizontalScale(
+            originLocation,
+            gridDimensions,
+            containerDiameter
+        );
+        // this scale is used for positioning the dies on the canvas
+        this._verticalScale = this.createVerticalScale(
+            originLocation,
+            gridDimensions,
+            containerDiameter
+        );
+        this._dieDimensions = {
+            width: Math.abs(
+                this._horizontalScale(0) - this._horizontalScale(1)
+            ),
+            height: Math.abs(this._verticalScale(0) - this._verticalScale(1))
+        };
+    }
+
+    private gridDimensionsValidAndDefined(): boolean {
+        return (
+            !this.wafermap.validity.invalidGridDimensions
+            && typeof this.wafermap.gridMinX === 'number'
+            && typeof this.wafermap.gridMinY === 'number'
+            && typeof this.wafermap.gridMaxX === 'number'
+            && typeof this.wafermap.gridMinX === 'number'
+        );
+    }
+
+    private calculateGridDimensionsFromBoundingBox(): GridDimensions {
+        const gridDimensions = { origin: { x: 0, y: 0 }, rows: 0, cols: 0 };
+        if (
+            typeof this.wafermap.gridMaxY === 'number'
+            && typeof this.wafermap.gridMinY === 'number'
+            && typeof this.wafermap.gridMaxX === 'number'
+            && typeof this.wafermap.gridMinX === 'number'
+        ) {
+            gridDimensions.origin.x = this.wafermap.gridMinX;
+            gridDimensions.origin.y = this.wafermap.gridMinY;
+            gridDimensions.rows = this.wafermap.gridMaxY - this.wafermap.gridMinY + 1;
+            gridDimensions.cols = this.wafermap.gridMaxX - this.wafermap.gridMinX + 1;
+        }
+        return gridDimensions;
+    }
+
+    private calculateGridDimensionsFromDies(): GridDimensions {
+        if (this.wafermap.diesTable === undefined) {
+            return { origin: { x: 0, y: 0 }, rows: 0, cols: 0 };
+        }
+
+        const colIndex = this.wafermap.diesTable
+            .getChild('colIndex')!
+            .toArray() as Int32Array;
+        const rowIndex = this.wafermap.diesTable
+            .getChild('rowIndex')!
+            .toArray() as Int32Array;
+
+        const minPoint = { x: colIndex[0]!, y: rowIndex[0]! };
+        const maxPoint = { x: colIndex[0]!, y: rowIndex[0]! };
+
+        // will replace iterating with arquero derive after fixing errors
```

Comment:
```suggestion
        // will replace iterating with arquero after fixing errors
```
Also, I think it's worth linking to the PR in arquero here.